state/application.go

```diff
@@ -1408,6 +1407,41 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	return nil
 }
 
+// MergeBindings merges the provided bindings map with the existing application
+// bindings.
+func (a *Application) MergeBindings(operatorBindings *Bindings, force bool) error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := a.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+
+		ch, _, err := a.Charm()
```

Comment:
Is there a mismatch in Force here?  SetCharm() uses the cfg.Force value with bindings.updateOps, here the force flag from the bind command is used with bindings.updateOps.  Is it possible for SetCharm() to fail later, because the force flag was needed here, and not saved?

Reply:
I don' think so. Running commands with `--force` should put the operator in a "trust me, I know what I 'm doing" mode. In my view, if the operator forces a bogus set of bindings with `juju bind` they shouldn't really be surprised if they also have to `--force` when upgrading a charm.

Reply:
okay.  And the corollary would be even if they forced the charm, the binding changes could be hazardous, so they should have to force the change independent of what the charm is set for.